assets/scripts/segments/__tests__/Segment.test.js

```diff
@@ -0,0 +1,94 @@
+/* eslint-env jest */
+import React from 'react'
+import { shallow } from 'enzyme'
+import { Segment } from '../Segment'
+import { getSegmentInfo, getSegmentVariantInfo } from '../info'
+import { infoBubble } from '../../info_bubble/info_bubble'
+import { SETTINGS_UNITS_IMPERIAL } from '../../users/constants'
+import {
+  SEGMENT_WARNING_OUTSIDE,
+  SEGMENT_WARNING_WIDTH_TOO_SMALL,
+  SEGMENT_WARNING_WIDTH_TOO_LARGE
+} from '../constants'
+
+jest.mock('../../app/routing')
+jest.mock('../info')
+jest.mock('../../streets/data_model', () => {})
+jest.mock('../../info_bubble/info_bubble')
+
+function connectDropTarget (el) { return el }
+function connectDragSource (el) { return el }
+const connectDragPreview = () => { }
+
+describe('Segment', () => {
+  const updateSegmentData = jest.fn()
+  let segment = {}
+  beforeEach(() => {
+    getSegmentInfo.mockResolvedValue('test')
+    getSegmentVariantInfo.mockResolvedValue({ name: 'test' })
+  })
+  describe('on mount', () => {
+    it('updates the left position for this segment', () => {
+      shallow(<Segment connectDropTarget={connectDropTarget} connectDragSource={connectDragSource} segment={segment} actualWidth={1} updateSegmentData={updateSegmentData} connectDragPreview={connectDragPreview} />)
+      expect(updateSegmentData).toHaveBeenCalledTimes(1)
+    })
+  })
+  describe('if the segment variant is updated', () => {
+    let updatePerspective
+    beforeEach(() => {
+      segment.variantString = 'variant'
+      updatePerspective = jest.fn()
+    })
+    describe('variant is different', () => {
+      it('updates perspective', () => {
+        const wrapper = shallow(<Segment connectDropTarget={connectDropTarget} connectDragSource={connectDragSource} segment={segment} actualWidth={1} updateSegmentData={updateSegmentData} connectDragPreview={connectDragPreview} updatePerspective={updatePerspective} />)
+        wrapper.setProps({ segment: { variantString: 'other' } })
+        expect(updatePerspective).toHaveBeenCalledTimes(4) // should probably only be 2
+      })
+    })
+    it('updates the left position for this segment', () => {
+      const wrapper = shallow(<Segment connectDropTarget={connectDropTarget} connectDragSource={connectDragSource} segment={segment} actualWidth={1} updateSegmentData={updateSegmentData} connectDragPreview={connectDragPreview} updatePerspective={updatePerspective} />)
+      wrapper.setProps({ segment: { variantString: 'other' } })
+      expect(updateSegmentData).toHaveBeenCalledTimes(7) // should probably only be 1
```

Comment:
These are interesting. So it looks like something is calling this multiple times, unnecessarily. Is it fine to leave here now as a reminder? Should we throw a TODO on this?

Reply:
yeah, we can leave it there and i add a TODO comment. 
this is probably because some props change and the check is done every time a prop is changed and `componentDidUpdate` gets called again.

Reply:
Sounds good!